### DISCUSSION TITLE
QQ是否已存在着能够直接识别非官方客户端登录的可能性？

### DISCUSSION THREAD

**Discussion by Amazefcc233** [Mrs4s/go-cqhttp#2225] [General]
先前有发过 #2161 的issue。服务器上挂了两个机器人且均会出现问题，但自己翻了翻依赖感觉问题正常情况不会存在，于是乎生成了若干个log文件，并**在自己的电脑上，使用了自己先前从未登录过gocqhttp的大号**尝试了一波，结果该bug无法复现，于是作罢并退出。整个登入时间加起来不会超过2分钟，且未进行过任何需主动请求的操作（如发消息等）。

结果大概过了3-5天，突然手机上收到“账号在其他地方上线，本地已下线”的通知，发现账号所有平台均已被强制登出。iPad端想再次上线时，直接被提示版本过低，要求强制升级。~~然后就迫不得已被用着改版之后的qq空间，难受死我了~~

这是我头一次遇到这种情况，qq本身就天天高强度水群/潜水也没碰到过。但在登了bot之后才出现这种情况，感觉很难说只是一种巧合，猜测着是否会有着某种手段能够直接识别是否为bot端登录的可能性...？

也希望我只是阴谋论罢了（瘫倒

**Comment by abersheeran** [comment]
默认生成的 device.json 名字都直接是 mirai，这不是直接告诉腾讯“我这个设备是 Mirai，速封我”。

**Reply by jinyu2022** [reply]
确实，那应该改成什么？

**Reply by misaka-link** [reply]
提取实机的复制进去呢?

**Reply by 5kind** [reply]
我是写了个脚本提取实机的信息

**Reply by landall** [reply]
实机信息长什么样子？

**Reply by 5kind** [reply]
我的脚本上传到github了，有点小问题不过我懒得改了，腾讯的检测还是得靠签名服务器过

**Comment by ilihu** [comment]
当然可以识别。例如，QQ登录需要签名，发消息需要签名。而之前大部分bot都是不带签名的，这不就是很明显的特征么
